### PR TITLE
fix #35147 QgsMeshLayer map canvas rotation rendering

### LINKED ISSUE #35147: [mesh] Problem with displaying mesh and canvas rotation
Peek 2020-03-17 19-55

### PR DESCRIPTION
**Pull request by PeterPetrik**
fix #35147

### CODE REVIEW

File: src/core/mesh/qgsmeshlayerutils.cpp [hunk 1]
```diff
@@ -197,20 +197,40 @@ QVector<double> QgsMeshLayerUtils::calculateMagnitudes( const QgsMeshDataBlock &
...
+  const QgsRectangle &bbox
+)
 {
-  QgsPointXY ll = mtp.transform( bbox.xMinimum(), bbox.yMinimum() );
-  QgsPointXY ur = mtp.transform( bbox.xMaximum(), bbox.yMaximum() );
-  topLim = std::max( int( ur.y() ), 0 );
-  bottomLim = std::min( int( ll.y() ), outputSize.height() - 1 );
-  leftLim = std::max( int( ll.x() ), 0 );
-  rightLim = std::min( int( ur.x() ), outputSize.width() - 1 );
+  QgsPointXY topLeft = mtp.transform( bbox.xMinimum(), bbox.yMaximum() );
```
Comment (vcloarec) [inline src/core/mesh/qgsmeshlayerutils.cpp:205]: const ?

File: src/core/mesh/qgsmeshlayerutils.cpp [hunk 2]
```diff
@@ -197,20 +197,40 @@ QVector<double> QgsMeshLayerUtils::calculateMagnitudes( const QgsMeshDataBlock &
...
+void QgsMeshLayerUtils::boundingBoxToScreenRectangle(
+  const QgsMapToPixel &mtp,
+  const QSize &outputSize,
+  const QgsRectangle &bbox,
+  int &leftLim,
+  int &rightLim,
+  int &bottomLim,
+  int &topLim )
+{
+  QgsRectangle screenBBox = boundingBoxToScreenRectangle( mtp, bbox );
```
Comment (vcloarec) [inline src/core/mesh/qgsmeshlayerutils.cpp:228]: const

File: src/core/mesh/qgsmeshlayerrenderer.cpp [hunk 3]
```diff
@@ -101,10 +101,9 @@ void QgsMeshLayerRenderer::calculateOutputSize()
   QgsRenderContext &context = *renderContext();
   QgsRectangle extent = context.mapExtent();
   QgsMapToPixel mapToPixel = context.mapToPixel();
-  QgsPointXY topleft = mapToPixel.transform( extent.xMinimum(), extent.yMaximum() );
-  QgsPointXY bottomright = mapToPixel.transform( extent.xMaximum(), extent.yMinimum() );
-  int width = int( bottomright.x() - topleft.x() );
-  int height = int( bottomright.y() - topleft.y() );
+  QgsRectangle screenBBox = QgsMeshLayerUtils::boundingBoxToScreenRectangle( mapToPixel, extent );
```
Comment (vcloarec) [inline src/core/mesh/qgsmeshlayerrenderer.cpp:104]: const

**Review by vcloarec — APPROVED** [review]

**Comment by nyalldawson** [comment]
Can we skip the backport here please?

**Comment by PeterPetrik** [comment]
@nyalldawson done

**Comment by PeterPetrik** [comment]
can I merge it now?

**Comment by nyalldawson** [comment]
Lgtm!

**Pull request merged**